ci/webdriver.py

```diff
@@ -0,0 +1,484 @@
+from selenium.webdriver.chrome.options import Options
+from selenium.webdriver.chrome.service import Service
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.ui import WebDriverWait
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "-a",
+    "--address",
+    dest="address",
+    default="localhost",
+    help="Address to testing Node server",
+)
+parser.add_argument(
+    "-p",
+    "--port",
+    dest="port",
+    default=3000,
+    type=int,
+    help="Port to testing Node server",
+)
+parser.add_argument(
+    "-k",
+    "--browser-kind",
+    dest="browser_kind",
+    choices=["chrome", "firefox"],
+    required=True,
+    help="Path to web driver file",
+)
+parser.add_argument(
+    "-b",
+    "--browser-path",
+    dest="browser_path",
+    required=True,
+    help="Path to browser file",
+)
+parser.add_argument(
+    "-d",
+    "--driver-path",
+    dest="driver_path",
+    required=True,
+    help="Path to web driver file",
+)
+parser.add_argument(
+    "--index-path",
+    dest="index_path",
+    default="tfhe/web_wasm_parallel_tests/index.html",
+    help="Path to HTML index file containing all the tests/benchmarks",
+)
+parser.add_argument(
+    "--id-pattern",
+    dest="id_filter_pattern",
+    help="Pattern to use to filter HTML button ID displayed on web page",
+)
+parser.add_argument(
+    "--value-pattern",
+    dest="value_filter_pattern",
+    help="Pattern to use to filter HTML button value displayed on web page",
+)
+parser.add_argument(
+    "-f",
+    "--fail-fast",
+    dest="fail_fast",
+    action="store_true",
+    help="Exit on first failed test",
+)
+parser.add_argument(
+    "--server-cmd",
+    dest="server_cmd",
+    help="Command to execute to launch web server in the background",
+)
+parser.add_argument(
+    "--server-workdir",
+    dest="server_workdir",
+    help="Path to working directory to launch web server",
+)
+
+
+class BrowserKind(enum.Enum):
+    """
+    Kind of browsers currently supported
+    """
+
+    chrome = 1
+    # firefox = 2
+
+
+class Driver:
+    """
+    Representation of a web driver relying on Selenium.
+    """
+
+    def __init__(self, browser_path, driver_path, browser_kind, threaded_logs=False):
+        """
+        :param browser_path: path to binary web browser as :class:`str`
+        :param driver_path: path to binary web driver as :class:`str`
+        :param browser_kind: :class:`BrowserKind`
+        :param threaded_logs: launch a thread to display log in parallel
+        """
+        self.browser_path = browser_path
+        self.driver_path = driver_path
+
+        self.browser_kind = browser_kind
+
+        self.options = Options()
+        self.options.binary_location = self.browser_path
+        self.options.add_argument("--headless")
+        if os.getuid() == 0:
+            # If user ID is root then driver needs to run in no-sandbox mode.
```

Comment:
a log here to indicate we have the flag, something like

"Script is running as root, running browser with --no-sandbox for compatibility" for example